ppy/osu
# Add missing mania tooltip overlay for 4k and 7k

Attempts to fix #30827 by adding osu!mania 4K/7K rank breakdowns to profile overlay tooltips, matching the behavior on the osu! website.

Changes:
- Added FourKey and SevenKey to CommonStrings for localization support
- Implemented Variant mapping in UserStatistics to properly handle 4K/7K data from the API 
- Added variant rank display logic in MainDetails for both global and country rank tooltips

The implementation follows the web version's data structure, using the variants field from the API response to show individual rankings for each key mode.

Testing:
- Hover over global rank in osu!mania profile to see 4K/7K breakdown
- Hover over country rank to see corresponding 4K/7K country rankings
